tests/unit/components/playback/controls/InputDropDown.spec.js

```diff
@@ -0,0 +1,218 @@
+import { shallowMount } from "@vue/test-utils";
+
+import Vuex from "vuex";
+import { createLocalVue } from "@vue/test-utils";
+import BootstrapVue from "bootstrap-vue";
+import "bootstrap/dist/css/bootstrap.min.css";
+
+let wrapper = null;
+
+const localVue = createLocalVue();
+localVue.use(Vuex);
+localVue.use(BootstrapVue);
+
+let NuxtStore;
+let store;
+
+describe("InputDropDown.vue", () => {
+  beforeAll(async () => {
+    // note the store will mutate across tests, so make sure to re-create it in beforeEach
+    const storePath = `${process.env.buildDir}/store.js`;
+    NuxtStore = await import(storePath);
+  });
+
+  beforeEach(async () => {
+    store = await NuxtStore.createStore();
+  });
+
+  afterEach(() => wrapper.destroy());
+  const nicknames = [
+    "Customer Account 1",
+    "Customer Account 2",
+    "Customer Account 3",
+  ];
+  test("When mounting InputDropDown from the build dist file, Then it loads successfully and the props defined title is rendered", () => {
+    const propsData = {
+      title_label: "Customer ID",
+      options_text: nicknames,
+    };
+    wrapper = shallowMount(DistComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    const target_span = wrapper.find(".span__input-dropdown-content-label");
+
+    expect(target_span.text()).toStrictEqual("Customer ID");
+  });
+  test("When the InputDropDown is mounted, Then it loads successfully and the props defined placeholder is rendered on input field", () => {
+    const propsData = {
+      title_label: "Customer ID",
+      options_text: nicknames,
+      placeholder: "Select Customer ID",
+    };
+    wrapper = mount(ComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    const target_input = wrapper.find("#input-dropdown-widget");
+    expect(target_input.attributes().placeholder).toStrictEqual(
+      "Select Customer ID"
+    );
+  });
+  test("When the InputDropDown is mounted, Then the user enters few charters in the input, confirm that an event 'update' is emitted with entered text", async () => {
+    const propsData = {
+      title_label: "Customer ID",
+      options_text: nicknames,
+      placeholder: "Select Customer ID",
+      value: "",
+    };
+    wrapper = mount(ComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    const customer = "New Customer"; // new Customer
+    const input_widget = wrapper.find("#input-dropdown-widget");
+    input_widget.element.value = customer;
+    await input_widget.trigger("input");
+    const parent_id_events = wrapper.emitted("update:value");
+    expect(parent_id_events).toHaveLength(1);
+    expect(parent_id_events).toStrictEqual([[customer]]);
+  });
+  test("When the InputDropDown is mouted, Then a prop message_if_blank is set to true and invalid_text is provided as value is <empty>, an error text is rendered", async () => {
+    const propsData = {
```

Comment:
please also assert that the `div__input-dropdown-controls-content-widget--invalid` class has been applied to the box, since that is also supposed to happen when it is invalid

Reply:
Now asserted for visibility.